src/intersections/lines.jl

```diff
@@ -48,7 +48,7 @@ calculated in order to identify the intersection type:
 function intersectparameters(a::Point{Dim,T}, b::Point{Dim,T}, 
                              c::Point{Dim,T}, d::Point{Dim,T}) where {Dim,T}
   A = [(b - a) (c - d)]
-  y = c - a
+  y = SVector(c - a)
```

Comment:
@ErickChacon can you comment again why this is necessary? StaticArrays.jl forces a conversion before A \ y? Would that be something to be fixed in StaticArrays.jl in the future?

Reply:
@juliohm, it is related to this line (https://github.com/JuliaArrays/StaticArrays.jl/blob/b4e49ce11fc88fc826b6353312f97e7101473509/src/solve.jl#L73).

```julia
similar_type(b, T, $outsize)(Matrix(a) \ b)
```

1. `a` gets converted to `Matrix`
2. solve operation is done using LinearAlgebra.jl
3.  finally, it try to convert the output (`Vector`) to the type of `b` (Vec).

So, step 3 does not work because we do not provide the method to convert an `AbstractVector` to `Vec`.

Reply:
And it works with vanilla `SVector`? I understood that `SVector` doesn't provide the method to convert `AbstractVector` to `SVector` either?

Reply:
They do not have non type-stable convertor, but they have a convertor that for a given `Type` and `size`. The function `similar_type` makes that conversion, but note that you need to provide `type` and `size` which they can know in advance because of matrix operations. We also have the constructor for a given Type and size, but not the non-type stable convertor.

Reply:
I don't know if I followed the difference between our custom `Vec` and their `SVector`. Are we missing some method implementation regarding this `similar_type` that would make the code work without the explicit conversion to `SVector`?

Reply:
Maybe we can use something simirlar to `similar_type`, I am not sure. I will have to check how that function work in `StaticArrays.jl`.

Reply:
There is some information here: https://github.com/JuliaArrays/StaticArrays.jl/blob/b4e49ce11fc88fc826b6353312f97e7101473509/docs/src/pages/api.md?plain=1#L99-L121

Reply:
@eliascarv since you were the one who originally looked into it, do you have an idea of what we would need to implement to support the A \ y without converting y to SVector explicitly?

Reply:
@juliohm I think the issue is fixed. It is actually not necessarily at all to modify this function. Let me explain:

1. Initially, I used to get the error for this function when I removed all `AbstractVector`  constructors meaning that there was no constructor like `Vec{Dim, T}(coords:AbstractVector)`, so I solved the problem using `y = SVector(c-a).
2. In other tests, I notice that it was necessary to use `Vec{Dim, T}(coords:AbstractVector)`, so I decided to put it back.
3. I did not notice that defining `Vec{Dim, T}(coords:AbstractVector)` helped `StaticArrays.jl` to work normally without the need of `SVector`.

Hence we do not need to change this function at all.